src/vlogger/vlogger.h

```diff
@@ -274,7 +277,10 @@ static inline void vlog_printf(vlog_levels_t log_level, const char* fmt , ... )
 	        if (len > VLOGGER_STR_SIZE - VLOGGER_STR_TERMINATION_SIZE) 
 	                len = VLOGGER_STR_SIZE - VLOGGER_STR_TERMINATION_SIZE - 1;
 
-		len += snprintf(buf+len, VLOGGER_STR_TERMINATION_SIZE, VLOGGER_STR_COLOR_TERMINATION_STR);
+		len = snprintf(buf + len, VLOGGER_STR_TERMINATION_SIZE, VLOGGER_STR_COLOR_TERMINATION_STR);
+		if (len < 0) {
```

Comment:
@igor-ivanov i expect we have this potential negative value returned also a few lines above where we handle the  'vsnprintf()'
Maybe we're better off using a temp ret_val. If negative exit. If positive add to 'len'.